src/gui/Src/Gui/StructWidget.cpp

```diff
@@ -467,7 +487,42 @@ void StructWidget::changeAddrSlot()
 
 void StructWidget::refreshSlot()
 {
-    typeUpdateWidget();
+    auto doRefresh = [&]
+    {
+        typeUpdateWidget();
+
+        // Update mInsertIndex to the current row index of the selected item
+        if(hasSelection)
+            mInsertIndex = ui->treeWidget->currentIndex().row();
+        else
+            mInsertIndex = -1; // Reset if no selection
+    };
+
+    doRefresh();
+
+    if(!hasSelection || selectedItem->parent() || !DbgIsDebugging())
+        return;
+
+    auto selectedAddr = selectedType.addr + selectedType.offset;

```

Comment:
Shouldn't the `selectedType.offset` always be equal to `0` (since only top-level types can be refreshed)?